### PR TITLE
feat: add product group page (draft)

### PR DESCRIPTION
**Pull request by ahmeterenodaci**
## How has this been tested?

- [ ] Prod
- [ ] Development

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] Added all translations, no hard coded text. 
- [ ] Updated the UI (if needed).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by ahmeterenodaci** [comment]
Bu sayfa baştan yazılacak hızlıca gösterebilmek için bu şekilde yapıldı

File: apps/web/src/app/[lang]/app/[type]/product-group/form.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        sections={[
+          { id: "default", name: "Ürün Grupları" },
+          { id: "vat", name: "Ürün Gruplarının KDV'leri" },
+        ]}
+      />
+      <div className="flex flex-col bg-white px-5">
+        <DataTable
+          columnsData={columnData}
+          data={productGroupData}
+          filterBy="result"
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/product-group/form.tsx:59]: The filter by should one of the inputs of the server for example: 
```suggestion
          filterBy="name"
```

File: apps/web/src/app/[lang]/app/[type]/product-group/loading.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                <div className="text-sm text-muted-foreground">
+                  <Skeleton className="h-4 w-full bg-gray-200 w-32" />
+                </div>
+              </div>
+              <div className="ml-auto">
+                <Skeleton className="h-8 mb-1 w-full bg-gray-200 w-20" />
+              </div>
+            </div>
+          ))}
+        </div>
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/product-group/loading.tsx:42]: We have a loading state inside the table itself we can use it.

File: apps/web/src/app/[lang]/app/[type]/product-group/page.tsx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+                <DropdownMenuItem>
+                  <Building2Icon className="mr-2 h-4 w-4" />
+                  <span>Kurumsal Profil</span>
+                </DropdownMenuItem>
+              </Link>
+            </DropdownMenuGroup>
+          </DropdownMenuContent>
+        </DropdownMenu>
+      </div> */}
+      <Form productGroupData={productGroupData} />
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/product-group/page.tsx:220]: Nit: Should we name this as table or data?

File: apps/web/src/app/[lang]/app/[type]/product-group/form.tsx
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/product-group/form.tsx:1]: we could change the file name to table or dashboard?

**Review by a0m0rajab — CHANGES_REQUESTED** [review]

File: apps/web/src/app/[lang]/app/[type]/product-group/loading.tsx (hunk 2 above)
Comment (ahmeterenodaci) [reply apps/web/src/app/[lang]/app/[type]/product-group/loading.tsx:43]: It can be used for data loading but not for page loading.

Comment (a0m0rajab) [reply apps/web/src/app/[lang]/app/[type]/product-group/loading.tsx:43]: @ahmeterenodaci we can use it for page loading.

**Review by a0m0rajab — APPROVED** [review]

**Pull request merged**